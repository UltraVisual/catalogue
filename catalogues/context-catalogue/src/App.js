import React  from 'react';
import  { Route } from 'react-router';
import  { Link } from 'react-router-dom';

import Header from './components/Header';
import Tiles from './components/Tiles';
import Detail from './components/Detail';
import './App.css';

const { Provider } = React.createContext('catalogue');

const App = () => {
  return (
    <div className="app">
        <Link to={ '/' }>
            <Header title={ 'Catalogue' }/>
        </Link>
        <Provider>
            <Route exact path="/" component={ () => <Tiles /> }/>
        </Provider>
        <Provider>
            <Route path="/detail/:name" component={ () => <Detail /> }/>
        </Provider>
    </div>
  )
};

export default App;
