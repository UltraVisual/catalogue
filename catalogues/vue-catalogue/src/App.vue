<template>
  <div id="app">
    <Header :title="title"/>
    <router-view/>
  </div>
</template>

<script>
import Header from '@/components/Header'

export default {
  name: 'App',
  components: {
    Header
  },
  data: () => ({
    title: 'Catalogue'
  })
}
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
