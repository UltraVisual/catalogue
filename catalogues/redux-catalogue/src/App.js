import React  from 'react';
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import  { Route } from 'react-router';
import  { Link } from 'react-router-dom';

import Header from './components/Header';
import Tiles from './components/Tiles';
import Detail from './components/Detail';
import rootReducer from './reducers';
import './App.css';

const store = createStore(rootReducer)

const App = () => {
  return (
    <div className="app">
      <Link to={ '/' }>
        <Header title={ 'Catalogue' }/>
      </Link>
      <Provider store={ store }>
        <Route exact path="/" component={ Tiles }/>
      </Provider>
      <Provider store={ store }>
        <Route path="/detail/:name" component={ Detail }/>        
      </Provider>
    </div>
  )
};

export default App;
