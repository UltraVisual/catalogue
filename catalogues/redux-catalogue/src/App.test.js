import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import { shallow } from 'enzyme';

it('renders without crashing', () => {
  let app = shallow(<App />);

  expect(app).toMatchSnapshot();
});
