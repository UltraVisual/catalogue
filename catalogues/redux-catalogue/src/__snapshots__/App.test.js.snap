// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
>
          <Header
                    title="Catalogue"
          />
</Link>,
        <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
</Provider>,
        <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
>
          <Route
                    component={[Function]}
                    path="/detail/:name"
          />
</Provider>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Header
            title="Catalogue"
/>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "Catalogue",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Route
            component={[Function]}
            path="/detail/:name"
/>,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/detail/:name",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
>
            <Header
                        title="Catalogue"
            />
</Link>,
          <Provider
            store={
                        Object {
                                    "dispatch": [Function],
                                    "getState": [Function],
                                    "replaceReducer": [Function],
                                    "subscribe": [Function],
                                    Symbol(observable): [Function],
                                  }
            }
>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
</Provider>,
          <Provider
            store={
                        Object {
                                    "dispatch": [Function],
                                    "getState": [Function],
                                    "replaceReducer": [Function],
                                    "subscribe": [Function],
                                    Symbol(observable): [Function],
                                  }
            }
>
            <Route
                        component={[Function]}
                        path="/detail/:name"
            />
</Provider>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Header
              title="Catalogue"
/>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "Catalogue",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Route
              component={[Function]}
              path="/detail/:name"
/>,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/detail/:name",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
